CMakeLists.txt

```diff
@@ -202,10 +202,6 @@ KOKKOS_SETUP_BUILD_ENVIRONMENT()
 # 7) Export and install targets
 
 OPTION(BUILD_SHARED_LIBS "Build shared libraries" OFF)
-# Workaround for building position independent code.
-IF(BUILD_SHARED_LIBS)
-  SET(CMAKE_POSITION_INDEPENDENT_CODE ON)
-ENDIF()
```

Comment:
This code was introduced in #2288.  We need to make sure removing these lines do not break support for shared libraries.
@junghans @masterleinad

Reply:
https://github.com/kokkos/kokkos/issues/4722#issuecomment-1022744946 I tested on gcc and Intel with shared libraries

Reply:
If it works now, I'm very happy without this workaround. It would be interesting what "fixed" it, though.

Reply:
Likely some subsequent change in how Kokkos defines its targets in CMake, so that the need for PIC actually gets picked up like it's supposed to. It would take combing through the combined history of Kokkos's build files and releases of CMake itself to pinpoint it, though.

Reply:
The workaround just implements the CMake default for shared libs, so from my side it is ok to drop that.